src/components/Text/index.tsx

```diff
@@ -81,40 +81,42 @@ type TextProps = {
   variant: TextVariant
   color?: Color
   prominence?: ProminenceProps
-  as: ElementType
+  as: keyof JSX.IntrinsicElements
   oneLine?: boolean
   disabled?: boolean
   italic?: boolean
   underline?: boolean
   id?: string
+  htmlFor?: string
 }
 
-const StyledText = styled('div', {
-  shouldForwardProp: prop =>
-    ![
-      'as',
-      'variant',
-      'color',
-      'prominence',
-      'oneLine',
-      'disabled',
-      'italic',
-      'underline',
-    ].includes(prop),
-})<{
-  color: Color
-  prominence: ProminenceProps
-  variant: TextVariant
-  oneLine: boolean
-  disabled: boolean
-  italic: boolean
-  underline: boolean
-}>(generateStyles)
+const StyledText = (as: keyof JSX.IntrinsicElements) =>
+  styled(as, {
+    shouldForwardProp: prop =>
+      ![
+        'as',
+        'variant',
+        'color',
+        'prominence',
+        'oneLine',
+        'disabled',
+        'italic',
+        'underline',
+      ].includes(prop),
+  })<{
+    color: Color
+    prominence: ProminenceProps
+    variant: TextVariant
+    oneLine: boolean
+    disabled: boolean
+    italic: boolean
+    underline: boolean
+  }>(generateStyles)
 
 const Text = ({
   variant,
   children,
-  as,
+  as = 'div',
```

Comment:
I assumed we always wanted to specify it ?

Reply:
@chambo-e You're right I was confused by control value in storybook